builder/format.go

```diff
@@ -0,0 +1,66 @@
+// Copyright © 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builder
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+
+	"github.com/clearlinux/mixer-tools/helpers"
+	"github.com/pkg/errors"
+)
+
+// UpdateFormatVersion updates the builder.conf file with a new format version
+func UpdateFormatVersion(b *Builder, version string) error {
+	b.Config.Swupd.Format = version
+	newver := "${1}" + b.Config.Swupd.Format
+	var re = regexp.MustCompile(`(FORMAT=)[0-9]*`)
+
+	builderData, err := ioutil.ReadFile(b.BuildConf)
+	if err != nil {
+		return errors.Wrap(err, "Failed to read builder.conf")
+	}
+
+	builderEdit := re.ReplaceAllString(string(builderData), newver)
+
+	builderOut := []byte(builderEdit)
+	if err = ioutil.WriteFile(b.BuildConf, builderOut, 0644); err != nil {
+		return errors.Wrap(err, "Failed to write new builder.conf")
+	}
+
+	return nil
+}
+
+// CopyFullGroupsINI copies the initial ini file which has ALL bundle definitions
+func CopyFullGroupsINI(b *Builder) error {
```

Comment:
I haven't thought carefully about how you're using these to see if it needs to be this way, so out of curiosity, why are all of these `func Function(b *Builder)` and thus called as `builder.Function(b)` down below, rather than `func (b *Builder) Function()` and `b.Function()`?

Reply:
So was there a reason for this? Since you're passing a builder object to all of them, you must have a builder object already by the time you call them; why aren't they just methods on the builder object, then?